### PR TITLE
DOC: Add a plugins section for extending simphony

### PR DESCRIPTION
**Pull request by itziakos**
This PR provides some basic documentation on contributing engine and visualization plugins

### CODE REVIEW

**Comment by itziakos** [comment]
@roigcarlo and @nathanfranklin please have a look

File: doc/source/plugins.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+Plugins
+=======
+
+The Simphony library can extended through two `entry points`_ for
+contributing python modules that container engine and visualization components:
+
+- ``simphony.engine`` -- A python module that provides one or more
+  classes that implement the :class:`ABCEngine` interface.
+
+- ``simphony.visualization`` -- A python module that provides a simple
```
Comment (nathanfranklin) [inline doc/source/plugins.rst]: ``simphony.visualization`` -> ``simphony.visualisation``

Comment (nathanfranklin) [inline doc/source/plugins.rst]: I see that currently ``simphony.visualisation`` for the plugin but the current module name is ``visualization`` (``simphony/visualization/``).


I prefer **visualization** but can imagine that **visualisation** might be preferred

File: doc/source/plugins.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+To declare that a package contains a visualization or engine modules
+for simphony, a developer has to add an entry point definition in the
+``setup.py`` of the contributing package.
+
+e.g.::
+
+    setup(
+       entry_points={
+        'simphony.engine': ['<name> = <module path>'])
```
Comment (nathanfranklin) [inline doc/source/plugins.rst]: module_path

File: doc/source/plugins.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
+Plugins
+=======
+
+The Simphony library can extended through two `entry points`_ for
```
Comment (nathanfranklin) [inline doc/source/plugins.rst]: SimPhoNy

File: doc/source/plugins.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+contributing python modules that container engine and visualization components:
+
+- ``simphony.engine`` -- A python module that provides one or more
+  classes that implement the :class:`ABCEngine` interface.
```
Comment (nathanfranklin) [inline doc/source/plugins.rst]: ABCEngine -> ABCModelingEngine

**Comment by nathanfranklin** [comment]
Looks great!

I just added a few minor comments regarding spellings

File: doc/source/plugins.rst (hunk 1 above)
Comment (itziakos) [inline doc/source/plugins.rst]: I have updated the rst, code and folders to use the uk spelling.

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**